gencred/README.md

```diff
@@ -146,6 +146,9 @@ users:
     client-key-data: fake-key-data
 ```
 
+#### Merging into a kubeconfig in a Kubernetes secret.
+If you store kubeconfig files in kubernetes secrets to allow pods to access other kubernetes clusters (like many of Prow's components require) consider using [`merge_kubeconfig_secret.py`](/gencred/merge_kubeconfig_secret.py) to merge the kubeconfig produced by `gencred` into the secret.
```

Comment:
That isn't how this script works. If we wanted to use `kubectl get` and `kubectl replace` like that we could use gencred directly where you used `merge_kubeconfig_sercret.py`. That doesn't satisfy our use case because we don't want to give the person running gencred (the cluster owner) access to our Prow service cluster.
The actual usage would be something like:
```
# Generate a kubeconfig.yaml as described and shown above, then run
./merge_kubeconfig_secret.py kubeconfig.yaml
```
I'll include a recipe.

> Edge cases (still not handled): replace how we access a cluster, remove access to a cluster

I don't think we've ever had reason to do this before? While that does sound potentially useful, right now I'm just trying to automate repetitive and error prone steps. Automating replacement and removal can't be done simply using the `KUBECONFIG` envvar and `kubectl config view` like I am using now so it doesn't seem worthwhile to try to include right now IMO.

Also note that currently this merges kubeconfig files back into the same secret key. Your example seems to assume that the original kubeconfig is read from some unspecified source key and the merged kubeconfig is written to the specified key. As written this script doesn't preserve the original kubeconfig file or any other secret keys.
It looks like `kubectl patch` should work with secrets so I think I can use `--source-key` and `--dest-key` flags and then only patch the destination key (which could be the same as the source). I'll make that change so we can suggest the pattern you are describing.